src/parquet/file/writer-internal.cc

```diff
@@ -142,7 +142,7 @@ ColumnWriter* RowGroupSerializer::NextColumn() {
   // Throws an error if more columns are being written
   auto col_meta = metadata_->NextColumnChunk();
 
-  if (current_column_writer_) { total_bytes_written_ = current_column_writer_->Close(); }
+  if (current_column_writer_) { total_bytes_written_ += current_column_writer_->Close(); }
```

Comment:
The fact that we are computing the total bytes written twice is a bad code smell to me -- that this bug was only caught because of the DCHECK in `RowGroupMetaDataBuilder::Finish` tells me that the unit tests elsewhere are inadequate. I would prefer to take out the the double-accounting (save these checks for for the unit test suite) and use the `total_bytes_written` passed as an argument to `Finish` but not until the tests are improved.